apps/ideaspace/src/components/modules/EditIdea/EditIdea.js

```diff
@@ -0,0 +1,192 @@
+  );
+
+  const [card, setCard] = React.useState({
+    ideaName: 'qq',
+    tagline: '',
+    description: '',
+    targetAudience: '',
+    features: '',
+    experience: '',
+    extraInfo: '',
+    involveLevel: 'zero',
+    hourCommitmentMin: 0,
+    hourCommitmentMax: 0,
+    difficultyLevel: 'Beginner',
+  });
+
+  React.useEffect(() => {
+    if (ideaId) {
+      axios.get(`${process.env.NEXT_PUBLIC_STRAPI_URL}/idea-cards/${ideaId}`)
+        .then(response => {
+          if (response.status === 200) {
+            if(userData.id !== 0){
+              if(response.data.author.id == userData.id){
+                setCard(response.data);
+              }else{
+                alert("This is not your idea. You can't edit it.");
+                window.history.back(-1);
+              }
+            }
+          }
+        })
+    }
+  }, [ideaId,userData.id]);
+
+  const SignupSchema = Yup.object().shape({
+    ideaName: Yup.string().required('Idea Name is Required.'),
+    description: Yup.string().required('Idea Description is Required.'),
+    experience: Yup.string().required('Experience is Required.'),
+    features: Yup.string().required('Idea Feature is Required.'),
+  });
+
+  const submitHandler = async (values) => {
+    //e.preventDefault();
+    if (values == card) {
+      alert("nothing chage");
+      return;
+    }
+    setSending(true);
+
+    const res = await axios.put(
+      `${process.env.NEXT_PUBLIC_STRAPI_URL}/idea-cards/${ideaId}`,
+      values
+    );
+
+    if (res.status === 200) {
+      setunsavedChanges(false);
+      if (await confirmNotice()){
+        setUrrl(`/ideaspace/workshop/${res.data.id}`);
+      }
+    } else {
+      alert('Unable to update your idea.');
+      setSending(false);
+      setunsavedChanges(true);
+    }
+  };
+
+  const handleDialog = async (url) => {
+    if (await confirmLeave()) {
+      setUrrl(url);
+    } else {
+      setSending(false);
+    }
+  }
+
+  React.useEffect(() => {
+    window.onbeforeunload = () => {
+      if (unsavedChanges) {
+        return 'You have unsaved changes. Do you really want to leave?';
+      }
+    };
+    
+    if (unsavedChanges && urrl == '' ) {
+      const routeChangeStart = (url) => {
+        handleDialog(url);
+        router.events.emit('routeChangeError');
+        throw 'Abort route change. Please ignore this error.';
+      }
+      router.events.on('routeChangeStart', routeChangeStart);
+      return () => {
+        router.events.off('routeChangeStart', routeChangeStart);
+      };
+    } else if (urrl !== '') {
+      if (urrl == 'back'){
+        window.history.back(-1);
+      }else{
+        router.push(urrl);
+      }
+    }
+  }, [unsavedChanges, urrl]);
+
+  const backHandler = (url) => {
+    if (unsavedChanges) {
+      handleDialog(url);
+    } else {
+      window.history.back(-1);
+    }
+  }
+
+  return (
+    <>
+      <HeadWapper>
+        <atoms.Layer hasRainbow style={{ width: '16.8rem', margin: 'auto' }}>
```

Comment:
@alicefate is there any way we could put these styles into a styled component file?

Reply:
we could design a new styled component by ourselves. Here is two ways to bring rainbow bar to title.  I use the first one when title is short, use the second one when title is long. Beacuse the first one will look strange in thin window with.
`
<atoms.Layer hasRainbowBottom style={{ width: '24.65rem', margin: 'auto' }}>
          <atoms.Typography type='h1' style={{ fontSize: '4rem', }}>
            Submit an idea
          </atoms.Typography>
        </atoms.Layer>

<Headline>IdeaSpace Dashboard</Headline>
        <StyledRanbow>
          <atoms.Layer hasRainbowBottom />
        </StyledRanbow>`